core/handlers/v2/simulation_handler.go

```diff
@@ -57,15 +59,25 @@ func (this *SimulationHandler) Put(w http.ResponseWriter, req *http.Request, nex
 
 	err := json.Unmarshal(body, &simulationView)
 	if err != nil {
-		handlers.WriteErrorResponse(w, err.Error(), http.StatusInternalServerError)
+
+		log.WithFields(log.Fields{
+			"body": string(body),
+		}).Debug(err.Error())
+
+		handlers.WriteErrorResponse(w, "Could not import simulation, did not follow valid v2 schema", http.StatusInternalServerError)
```

Comment:
So it was kinda doing this, but I wasn't outputting the right error messages. Fixed up so now it will report if the schema was wrong or if the JSON was invalid. Changed a test and added a new test to these both these errors.